src/Symfony/Component/Form/PreloadedExtension.php

```diff
@@ -44,9 +44,16 @@ class PreloadedExtension implements FormExtensionInterface
      */
     public function __construct(array $types, array $typeExtensions, FormTypeGuesserInterface $typeGuesser = null)
     {
-        $this->types = $types;
         $this->typeExtensions = $typeExtensions;
         $this->typeGuesser = $typeGuesser;
+
+        foreach ($types as $name => $type) {
+            // Up to Symfony 2.8, types were identified by their names
+            $this->types[$name] = $type;
```

Comment:
We already did expect people to pass an array with the type names as indices here, so I don't want to change that. Realistically, this class is only used inside `FormFactoryBuilder` anyway.